client/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+const ProtectedRoute = ({ path, component: Component, ...rest }: { path: string; component: any }) => {
+  const loggedInUser = useAuth();
```

Comment:
Also, consider destructuring this, as the useAuth hook returns an object and we only need access to a single piece of state (i.e loggedInUser in this case)

Reply:
destructuring added for "loggedInUser".